Controller/LoginController.php

```diff
@@ -93,7 +93,7 @@ public function connectAction(Request $request): Response
         // if connecting is enabled and there is no user, redirect to the registration form
         if ($this->connect && !$hasUser && $error instanceof AccountNotLinkedException) {
             $key = time();
-            $session = $request->hasSession() ? $request->getSession() : $this->session;
+            $session = $request->hasSession() ? $request->getSession() : $this->requestStack->getSession();
```

Comment:
As of minimal version of symfony components is 4.4 here, you cannot use `RequestStack::getSession()` here as it is introduced in 5.3

4.4 version - https://github.com/symfony/symfony/blob/v4.4.25/src/Symfony/Component/HttpFoundation/RequestStack.php
5.3 version -
https://github.com/symfony/symfony/blob/v5.3.2/src/Symfony/Component/HttpFoundation/RequestStack.php#L119-L131

Reply:
@andrew-demb Thank you for your comment! I was waiting for the CI to run to be warned of these kind of issues with composer `--prefer-lowest`.

I'll change it for `getCurrentRequest()->getSession()` if that's OK?

Reply:
I think yes, but we should handle request without session somehow.

Symfony 4 have a trigger deprecation [1] in this case (calling `getSession` without a session and return null, which is not expected here).

[1] https://github.com/symfony/symfony/blob/v4.4.25/src/Symfony/Component/HttpFoundation/Request.php#L718

Reply:
What should we do? Always throw an exception when there is no session? What happens today in this case? Do we get an exception when `SessionInterface` is autowired, or is it guaranteed to be available?

Also, I guess that if we have to introduce some logic to get the session (basically mimicking `RequestStack::getSession()`), we'll have to factorize it somewhere. Where? Something like a `SessionService`?

@XWB Would that be acceptable?

Reply:
A private method inside the controller would be fine, we made similar BC hacks in the past :)

Reply:
@XWB Thanks, I will try an implementation in this direction. Can you please approve the workflow so that I can see if the CI is successful?

Reply:
Yes, go for it :)